src/NServiceBus.Transport.RabbitMQ/Connection/ChannelProvider.cs

```diff
@@ -19,6 +21,35 @@ public ChannelProvider(ConnectionFactory connectionFactory, IRoutingTopology rou
         public void CreateConnection()
         {
             connection = connectionFactory.CreatePublishConnection();
+            connection.ConnectionShutdown += Connection_ConnectionShutdown;
+        }
+
+        void Connection_ConnectionShutdown(object sender, ShutdownEventArgs e)
+        {
+            if (e.Initiator != ShutdownInitiator.Application)
+            {
+                Task.Run(Reconnect).Ignore();
```

Comment:
This is risky as we allow potentially multiple `Reconnects` happening at the same time. If the newly created connection shutsdown immediately we have two threads in the `Reconnect`.

I suggest creating a never ending `Reconnect` task and make `Connection_ConnectionShutdown` set bool flag to indicate if reconnection is needed + unsubscribe from the handler (all in lock statement).

Reply:
> This is risky as we allow potentially multiple Reconnects happening at the same time. If the newly created connection shutsdown immediately we have two threads in the Reconnect.

I think we'll need to chat more about this, because I'm not seeing the problem.

Reply:
Discussed this with @tmasternak and the concern was around the `reconnected` value, but since that is a local variable, there should be no problem.